src/libmeasurement_kit/dns/resolver.hpp

```diff
@@ -0,0 +1,152 @@
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#include <measurement_kit/common.hpp>
+#include <measurement_kit/dns.hpp>
+
+#include <thread>
+/// XXX use versions compatible with windows
+#include <netdb.h>
+#include <sys/socket.h>
+#include <sys/types.h>
```

Comment:
The correct way to go about portability, in my opinion, is the following: you should make sure all Unix-specific headers you want to include are in `<measurement_kit/portable/>` and that they're pulled by `<measurement_kit/aaa_base.hpp>`. Then, you need to check what Windows headers are required to achieve a similar task and &mdash; when we'll merge the Windows patch &mdash; we'll make sure that those headers are also listed and pulled by `aaa_base.hpp` by using `#ifdef WIN32` when required (from a cursory analysis it seems to me that `netdb.h` is not part of `portable` headers and that `winsock2.h` is all you need on Windoze to call getaddrinfo).

As a side note, I also see the possibility of tapping into libevent's `getaddrinfo()` but I'd rather not do so as this level because it seems to me that libevent's `getaddrinfo()` only concerns itself with portability towards old systems of which we do not care about, plus I'd rather keep the surface of code depending on libevent small (and this is the rationale behind having a `libevent` folder).